src/main/java/core/basesyntax/Lottery.java

```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery {
+    public Ball getRandomBall() {
+        Random randomBall = new Random();
+        ColorSupplier colorSupplier = new ColorSupplier();
+        Ball ball = new Ball();
+        ball.setColor(colorSupplier.getRandomColor());
+        ball.setNumber(randomBall.nextInt(100));

```

Comment:
Each time this method is called, new instances of ColorSupplier and Random are created. Let's think about how we can optimize this logic and create each instance only once.